### PR TITLE
Add support for Iceberg storage in GCS

### PR DESCRIPTION
**Pull request by loicalleyne**
Add support for Iceberg storage in GCS.

Tested that it works with my setup of Polaris REST catalog with GCS storage.

### CODE REVIEW

File: lake_viewer.py
Comment (rakeshJn) [inline lake_viewer.py:1]: Actually, pyiceberg is good at reading env variables if defined properly, so we don't really need any of these here. Please see: https://py.iceberg.apache.org/configuration/#catalogs

File: lake_viewer.py [hunk 1]
```diff
@@ -15,15 +15,33 @@
...
+                    })
+            case _: 
+                self.catalog = catalog.load_catalog("default", 
+                    **{
+                        'uri': os.environ.get("PYICEBERG_CATALOG__DEFAULT__URI"),
+                        'token': os.environ.get("PYICEBERG_CATALOG__DEFAULT__TOKEN"),
+                        's3.endpoint':  os.environ.get("AWS_ENDPOINT"),
+                        'py-io-impl':   'pyiceberg.io.fsspec.FsspecFileIO',
+                        'warehouse': os.environ.get("PYICEBERG_CATALOG__DEFAULT__WAREHOUSE"),                
+                    })
```
Comment (rakeshJn) [inline lake_viewer.py:44]: Actually, pyiceberg is good at reading env variables if defined properly, so we don't really need any of these here. Please see: https://py.iceberg.apache.org/configuration/#catalogs

**Comment by loicalleyne** [comment]
@rakeshJn 
I cleaned up the code as per your suggestions, could you take a look please?
Cheers

File: my.env [hunk 2]
```diff
@@ -1,7 +1,10 @@
+GCP_KEYFILE=
 PYICEBERG_CATALOG__DEFAULT__URI=
 PYICEBERG_CATALOG__DEFAULT__TOKEN=
 PYICEBERG_CATALOG__DEFAULT__WAREHOUSE=
```
Comment (rakeshJn) [inline my.env:4]: Since this is removed from code, we need to define it in env.
PYICEBERG_CATALOG__DEFAULT__PY_IO_IMPL=pyiceberg.io.fsspec.FsspecFileIO

**Pull request merged**